visualization/app/codeCharta/util/fileValidator.ts

```diff
@@ -22,23 +69,45 @@ export class FileValidator {
 		return true
 	}
 
-	public static validate(file: { nodes: CodeMapNode[] }): Array<{ message: string; dataPath: string }> {
-		if (!file) {
-			return [{ message: "file is empty or invalid", dataPath: "empty or invalid file" }]
+	private static checkApiVersion(file: { nodes: CodeMapNode[] }) {
+		let apiVersion = this.getCodeChartaApiVersion()
+		let fileApiVersion = this.getFileApiVersion(file)
+
+		let apiVersionMajorMinor = [false, false]
+
+		let apiVersionMajor = apiVersion.split(".")[0]
+		let apiVersionMinor = apiVersion.split(".")[1]
+
+		let majorVersion = fileApiVersion.split(".")[0]
+		let minorVersion = fileApiVersion.split(".")[1]
+
+		if (majorVersion > apiVersionMajor) {
+			apiVersionMajorMinor[0] = true
+		} else if (minorVersion > apiVersionMinor) {
+			apiVersionMajorMinor[1] = true
 		}
-		let ajv = require("ajv")()
-		let compare = ajv.compile(require("./schema.json"))
-		let valid = compare(file)
+		return apiVersionMajorMinor
+	}
 
-		if (!FileValidator.hasUniqueChildren(file.nodes[0])) {
-			return [
-				{
-					message: "names or ids are not unique",
-					dataPath: "uniqueness"
+	private static getCodeChartaApiVersion() {
+		let schemaCodeCharta = codeCharta
+		let schemaApiVersion
+
+		for (let key in schemaCodeCharta) {
+			if (key === "codecharta") {
+				schemaApiVersion = schemaCodeCharta[key]
+				for (let apiVersionKey in schemaApiVersion) {
+					return schemaApiVersion[apiVersionKey]
 				}
-			]
+			}
 		}
+	}
```

Comment:
This method is somehow super complex for the outcome. Isn't the returned value for an apiVersion just the major version like below?
- `1.2` -> `"1"`
- `3.4` -> `"3"`

Instead of checking if the apiVersion is inside the package.json, I would rather use a simple variable to get the major version ...
```
const latestMajorApiVersion = require("../../../package.json").codecharta.apiVersion[0]
```
... and add a unit test which checks that the property is available inside the package.json and matches a valid apiVersion.